scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -433,21 +424,86 @@ public JobId submit(Job userJob)
                 throw new SubmissionClosedException(msg);
             }
 
-            UserIdentificationImpl ident = frontendState.checkPermission("submit",
-                                                                         YOU_DO_NOT_HAVE_PERMISSION_TO_SUBMIT_A_JOB);
+            final UserIdentificationImpl ident = frontendState.checkPermission("submit",
+                                                                               YOU_DO_NOT_HAVE_PERMISSION_TO_SUBMIT_A_JOB);
 
-            InternalJob job = frontendState.createJob(userJob, ident);
+            final InternalJob job = frontendState.createJob(userJob, ident);
 
-            schedulingService.submitJob(job);
-
-            frontendState.jobSubmitted(job, ident);
-            return job.getId();
+            final Future<?> submitting = submitPool.submit(() -> {
+                try {
+                    schedulingService.submitJob(job);
+                    frontendState.jobSubmitted(job, ident);
+                } catch (Exception e) {
+                    if (job.getId().longValue() == InternalJob.UNINITIALIZED_JOB_ID) {
+                        logger.error(String.format("Error while submitting job %s. JobID wasn't available yet.",
+                                                   userJob.getName()),
+                                     e);
+                    } else {
+                        logger.error(String.format("Error while submiting a job[id:%s] ", job.getId().value()), e);
+                    }
+                    throw new RuntimeException(e);
+                } finally {
+                    unregisterSumbittedJob(job.getId());
+                    ;
+                }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)